### PR TITLE
Add "objective":"maximize" to metrics

### PR DESCRIPTION
**Pull request by lydiaxing**
In light of metric minimization, it is now best practice to include `"objective": "maximize"` for all metrics. This PR adds a `metrics` field (so all metrics in the examples are now named) to all experiment create calls and adds `"objective": "maximize"` (or `"objective": "minimize"`, if appropriate) to all metrics.

### CODE REVIEW

File: conditionals/python/conditionals.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
       values=['gaussian1', 'gaussian2'],
     )
   ],
+  metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline conditionals/python/conditionals.py]: maybe name could be `score` instead of `value` to not confuse it with `values`

File: constraints/python/constraints.py [hunk 2]
```diff
@@ -20,6 +20,7 @@
     dict(name="x", type="double", bounds=dict(min=-1, max=2)),
     dict(name="y", type="double", bounds=dict(min=-1, max=1)),
   ],
+  metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline constraints/python/constraints.py]: `score` instead of `value` here as well

File: dnn-tuning-nervana/nervana_all_cnn.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 experiment = conn.experiments().create(
   name='Nervana All CNN GPU '+datetime.datetime.now().strftime("%Y_%m_%d_%I%M_%S"),
   project='sigopt-examples',
+  metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline dnn-tuning-nervana/nervana_all_cnn.py]: score instead of value here as well

File: dnn-tuning-nvidia-mxnet/multi_run_experiments.py [hunk 4]
```diff
@@ -40,7 +40,10 @@
         project='sigopt-examples',
         parameters=hyperparams,
         observation_budget=30 * len(hyperparams),
-        metrics=[{'name': 'accuracy'}, {'name': 'negative_train_time'}],
+        metrics=[
+            {'name': 'accuracy', 'objective': 'maximize'},
+            {'name': 'train_time', 'objective': 'minimize'},
+        ],
```
Comment (oliviakim95) [inline dnn-tuning-nvidia-mxnet/multi_run_experiments.py]: 🎉

File: dnn-tuning-nvidia-mxnet/run_experiments.py [hunk 5]
```diff
@@ -35,6 +35,7 @@
     experiment = conn.experiments().create(
                          name=exp_name,
                          project='sigopt-examples',
+                         metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline dnn-tuning-nvidia-mxnet/run_experiments.py]: score instead of value here as well

File: jupyter-notebook-example/SigOpt_Introduction.ipynb [hunk 6]
```diff
@@ -84,6 +84,7 @@
     "experiment = conn.experiments().create(\n",
     "  name='SigOpt Introduction Example',\n",
     "  project='sigopt-examples',\n",
+    "  metrics=[dict(name='value', objective='maximize')],\n",
```
Comment (oliviakim95) [inline jupyter-notebook-example/SigOpt_Introduction.ipynb]: score instead of value here as well

File: multimetric-demo/finding_multiple_optima.ipynb [hunk 7]
```diff
@@ -90,7 +90,7 @@
     "multimodal_first_solution_meta = {\n",
     "    'name': 'SigOpt Multimetric Demo - Single Metric Optimization (python)',\n",
     "    'project': 'sigopt-examples',\n",
-    "    'metrics': [{'name': 'value'}],\n",
+    "    'metrics': [{'name': 'value', 'objective': 'maximize'}],\n",
```
Comment (oliviakim95) [inline multimetric-demo/finding_multiple_optima.ipynb]: hmm looks like we have used `value` in the past... seems like we should rename this

Comment (lydiaxing) [reply multimetric-demo/finding_multiple_optima.ipynb]: I agree, going further, we can rename all of the `value` metrics to `score` instead?

File: random-forest/python/random_forest.py [hunk 8]
```diff
@@ -25,6 +25,7 @@
 experiment = conn.experiments().create(
   name="Random Forest (Python)",
   project="sigopt-examples",
+  metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline random-forest/python/random_forest.py]: `accuracy` instead of `value`?

File: reinforcement-learning/dqn.py [hunk 9]
```diff
@@ -23,6 +23,7 @@ def main():
     name= ENVIRONMENT_NAME + ' (DQN)',
     project='sigopt-examples',
     observation_budget=128,
+    metrics=[dict(name='value', objective='maximize')],
```
Comment (oliviakim95) [inline reinforcement-learning/dqn.py]: `average_reward` instead of `value`

File: multimetric-demo/finding_multiple_optima.ipynb (hunk 7 above)
Comment (oliviakim95) [reply multimetric-demo/finding_multiple_optima.ipynb]: hmm I think in some cases there are actual metrics that are being optimized, like `accuracy`. I think it'd be best to infer from the example what the objective metric and use that? In some examples they just use a function evaluation which I think would be suitable to call `score`

File: amazon-ml-hyperparameter-optimization/README.md [hunk 10]
```diff
@@ -58,7 +58,7 @@ The [hyperparameters](http://docs.aws.amazon.com/machine-learning/latest/dg/trai
...
 
-Hyperparameter optimization works to maximize some performance metric of your model. This example maximizes the Area Under the (Receiver Operating Characteristic) Curve (AUC) of a binary classifier because it is easily exposed by Amazon Machine Learning. Alternatives include maximizing accuracy or minimizing error (maximizing negative error). Read more about how Amazon measures model performance on [Binary Model Insights](http://docs.aws.amazon.com/machine-learning/latest/dg/binary-model-insights.html).
+Hyperparameter optimization works to maximize some performance metric of your model. This example maximizes the Area Under the (Receiver Operating Characteristic) Curve (AUC) of a binary classifier because it is easily exposed by Amazon Machine Learning. Alternatives include maximizing accuracy or minimizing error. Read more about how Amazon measures model performance on [Binary Model Insights](http://docs.aws.amazon.com/machine-learning/latest/dg/binary-model-insights.html).
```
Comment (oliviakim95) [inline amazon-ml-hyperparameter-optimization/README.md]: "Hyperparameter optimization works to `optimize` some performance metric of your model."?

File: multimetric-timeseries/train.py [hunk 11]
```diff
@@ -157,7 +157,7 @@ def evaluate_assignments(experiment, suggestion,
     inference_time = calculate_inference_time(model, x_test)
 
     metrics = [dict(name='val_acc', value=log.iat[-1, -2]),
-               dict(name='negative_inference_time', value=-inference_time)]
+               dict(name='inference_time', value=-inference_time)]
```
Comment (oliviakim95) [inline multimetric-timeseries/train.py]: shouldn't we change this to `dict(name='inference_time', value=inference_time)`

File: orchestrate/tensorflow_mnist/model.py [hunk 12]
```diff
@@ -143,8 +143,8 @@ def evaluate_model(model, x_test, y_test):
     sess.run(tf.global_variables_initializer())
     train_start = time()
     train_model(model, x_train, y_train)
-    orchestrate.io.log_metric('negative_train_time', train_start - time())
+    orchestrate.io.log_metric('train_time', train_start - time())
```
Comment (oliviakim95) [inline orchestrate/tensorflow_mnist/model.py]: `time() - train_start`

File: orchestrate/tensorflow_mnist/model.py [hunk 13]
```diff
@@ -143,8 +143,8 @@ def evaluate_model(model, x_test, y_test):
...
     eval_start = time()
     accuracy = evaluate_model(model, x_test, y_test)
-    orchestrate.io.log_metric('negative_evaluation_time',  eval_start - time())
+    orchestrate.io.log_metric('evaluation_time',  eval_start - time())
```
Comment (oliviakim95) [inline orchestrate/tensorflow_mnist/model.py]: `time() - eval_start`

File: multimetric-demo/svm_example.ipynb [hunk 14]
```diff
@@ -199,7 +199,10 @@
     "svm_score_time_meta = {\n",
     "    'name': 'SigOpt Multimetric Demo - Balancing Score vs. Training Time (python)',\n",
     "    'project': 'sigopt-examples',\n",
-    "    'metrics': [{'name': 'score'}, {'name': 'training time'}],\n",
+    "    'metrics': [\n",
+    "        {'name': 'score'},\n",
```
Comment (oliviakim95) [inline multimetric-demo/svm_example.ipynb]: I think these are missing the objectives

Comment (oliviakim95) [reply multimetric-demo/svm_example.ipynb]: also since `training time` is negated, will have to update the observation values to not be negative

Comment (oliviakim95) [reply multimetric-demo/svm_example.ipynb]: also there's another mention of negating here `https://github.com/sigopt/sigopt-examples/pull/127/files#diff-3c62cc8701b9e60c3e2fd912f2421925R181`
